testing/test_dataset.py

```diff
@@ -1424,6 +1517,57 @@ def testArrayMinMeanMax(self):
         # Run tests
         np.testing.assert_array_equal(result, expected)
 
+    @pytest.mark.parametrize('xform',
+                             [False, True],
+                             ids=['Original Values', 'XFormed Values'])
+    def testArrayMMMShape(self, S5E25D40Shapes, S5E25D40MMMs, XformS5E25D40MMMs, xform):
+        """ Utility method to check that the given dict of MMMs has the right shape """
+        accShapes = {chID: np.shape(MMMArr) for chID, MMMArr in (XformS5E25D40MMMs if xform else S5E25D40MMMs).items()}
+        assert accShapes == S5E25D40Shapes
+
+    @pytest.mark.parametrize('channels, xform',
+                             [((8, 80), False),
+                              ((84, 20, 59, 76), False),
+                              ((8, 80), True),
+                              ((84, 59, 76), True)],
+                             ids=['Original MMM Encoded', 'Original Non-Encoded',
+                                  'XFormed MMM Encoded', 'XFormed Non-Encoded'])
+    def testArrayMMMComparisons(self, S5E25D40MMMs, XformS5E25D40MMMs, channels, xform):
+        """ Utility method to check that mins < means < maxes """
+
+        MMMs = {chID: MMM for chID, MMM in (XformS5E25D40MMMs if xform else S5E25D40MMMs).items() if chID in channels}
+
+        failedAsserts = []
+        for chID, MMMArr in MMMs.items():
+            if np.greater(MMMArr[0][1:], MMMArr[1][1:]).any():
+                failedAsserts.append(f"CH{chID} FAILED MIN < MEAN\nMIN:\n{MMMArr[0][1:]}\nMEAN:\n{MMMArr[1][1:]}")
+            if np.greater(MMMArr[0][1:], MMMArr[2][1:]).any():
+                failedAsserts.append(f"CH{chID} FAILED MEAN < MAX\nMEAN:\n{MMMArr[1][1:]}\nMAX:\n{MMMArr[2][1:]}")
+
+        assert not failedAsserts, "\n".join(failedAsserts)
```

Comment:
That would be fine, I had that initially but my only concern was that if one channel fails this test, all other channels following it won't be checked since the loop terminates at the first fail. Could it ever be useful to have the list of all the channels that failed?

Reply:
I see what you mean. In that case, I'd test each channel individually, as opposed to one call that tests both 8 and 80. The channels are largely independent of one another.